### PR TITLE
Add lsp_buffer_enabled

### PR DESCRIPTION
**Pull request by mattn**
Current implementation of vim-lsp fire lsp_server_init event after received response of initialize method. So

1. `:e foo.rb`
2. `:new` immediately
3. waiting

lsp_server_init event is triggered on empty buffer. I want to do key-mappings only to buffers with LSP enabled.

```vim
function! s:lsp_install() abort
  setlocal omnifunc=lsp#complete
  setlocal signcolumn=yes
  nmap <buffer> gd <plug>(lsp-definition)
  nmap <buffer> <f2> <plug>(lsp-rename)
endfunction

augroup lsp_install
  au!
  autocmd User lsp_server_init call s:lsp_install()
augroup END
```

But lsp_server_init event is not useful for this.  So I added new event lsp_enabled. This event is triggered imediately when the buffer is currently active. If the buffer is not current active, the event will be triggered when the buffer will be active.

This will be merged, we can write settings like below.
```vim
function! s:lsp_install() abort
  setlocal omnifunc=lsp#complete
  setlocal signcolumn=yes
  nmap <buffer> gd <plug>(lsp-definition)
  nmap <buffer> <f2> <plug>(lsp-rename)
endfunction

augroup lsp_install
  au!
  autocmd User lsp_enabled call s:lsp_install()
augroup END
```

### CODE REVIEW

**Comment by prabirshrestha** [comment]
This is something I have always wanted. Can we rename the event to lsp_buffer_enabled So it doesn’t confuse with the entire Lsp enable/disable.
We should document this as I would see lot of people using this.

**Comment by mattn** [comment]
Updated.

FYI, currently, I'm working on https://github.com/mattn/vim-lsp-settings

This will be a collection of configurations for vim-lsp. If user does not have the language server, this plugin suggest to install them.

File: doc/vim-lsp.txt [hunk 1]
```diff
@@ -131,6 +139,22 @@ on pyls (https://github.com/palantir/python-language-server)
 <
     For more details refer to |lsp#register_server()|.
 
+3. Configure your settings for the buffer
+   Use |lsp_buffer_enabled| autocommand to configure the buffer.
+>
+        function! s:lsp_install() abort
```
Comment (prabirshrestha) [inline doc/vim-lsp.txt]: Might be rename this to on_lsp_buffer_enabled?

**Comment by prabirshrestha** [comment]
Not comment. Feel free to merge.

Seems like similar to this.
https://github.com/pdavydov108/vim-lsp-cquery
https://github.com/ryanolsonx/vim-lsp-typescript

I’m more than happy to refer vim-lsp-settings in README.md and vim-lsp docs. It would make it really easy for new users as well as existing users to benefit from language server.

**Comment by mattn** [comment]
Thanks.

File: doc/vim-lsp.txt [hunk 2]
```diff
@@ -151,7 +151,7 @@ on pyls (https://github.com/palantir/python-language-server)
 
        augroup lsp_install
             au!
-            autocmd User lsp_enabled call s:lsp_install()
+            autocmd User lsp_enabled call s:on_lsp_buffer_enabled()
```
Comment (prabirshrestha) [inline doc/vim-lsp.txt]: Should be lsp_buffer_enabled

Comment (mattn) [reply doc/vim-lsp.txt]: Yes, will merge https://github.com/prabirshrestha/vim-lsp/pull/601

**Pull request merged**